_im-plugin/index-context.md

```diff
@@ -0,0 +1,170 @@
+---
+layout: default
+title: Index context
+nav_order: 14
+redirect_from:
+  - /opensearch/index-context/
+---
+
+# Index context
+
+Context can be used to declare the use case for which you are building your index. 
+Using the context information, OpenSearch can apply a pre-determined set of settings, and mappings to do the heavy-lifting around ensuring:
+
+1. Available performance optimizations are applied for the index.
+2. Existing settings are tuned according to the use-case.
+3. Mappings/Aliases based on [OpenSearch Integrations]({{site.url}}{{site.baseurl}}/integrations/)  are applied on the index if applicable.
+
+These metadata parameters applied are backed by component templates which are automatically loaded when your cluster comes up. These component templates are prefixed with `@abc_template@`, and can only be used based on the usage pattern outlined in this document.
+{: .important}
+
+The `context` parameter is available on indices and index templates as an experimental feature starting OpenSearch 2.17
+{: .warning}
+
+## Usage
+
+### Installation
+
+In order to use the `context` field while creating an index or index-template:
+
+1. Install the `opensearch-system-templates` plugin on all nodes in your cluster using one of the [installation methods]({{site.url}}{{site.baseurl}}/install-and-configure/plugins/#install)
+2. Set the feature flag `opensearch.experimental.feature.application_templates.enabled` to `true`. For more information about enabling and disabling feature flags, see [Enabling experimental features]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/experimental/).
+3. Enable setting `cluster.application_templates.enabled` to `true`. See [configuring settings]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/index/#static-settings) for steps on how to do this.
```

Comment:
@mgodwan do we need 2 settings? experimental and cluster setting?

Reply:
Yes, the other setting is to ensure that once feature flag is removed, users still have a way to opt-in/out of using this